### PR TITLE
add page route animation article to cookbook

### PR DESCRIPTION
**Pull request by johnpryan**
This adds the animation article to the cookbook.

@sfshaza2 the cookbook index lists articles alphabetically, but concept-wise I think this should be the third article. Let me know what you think and I can revert 7310a273e7bf10dc05fa77cec28dd6a1e3db7358

FYI @RedBrogdon

### CODE REVIEW

File: src/docs/cookbook/animation/page-route-animation.md [hunk 1]
```diff
@@ -0,0 +1,277 @@
...
+  title: Animate the properties of a container
+  path: /docs/cookbook/animation/animated-container
+---
+
+# Animate a page route transition
+
+A design language, such as Material, defines standard behaviors when
+transitioning between routes (or screens). Sometimes, though, a custom
+transition between screens can make an app more unique. To help,
+[PageRouteBuilder](https://api.flutter.dev/flutter/widgets/PageRouteBuilder-class.html)
```
Comment (sfshaza2) [inline src/docs/cookbook/animation/page-route-animation.md]: Generally, we use the following format for links. (We didn’t always do that, but it’s now part of our recommended guidelines.) Also, we use macros for the common websites. So:

To help, [PageRouteBuilder] provides an [Animation]…

[PageRouteBuilder]:  {{site.api}}/flutter/widgets/PageRouteBuilder-class.html
[Animation]: {{site.api}}/flutter/widgets/PageRouteBuilder-class.html

It makes the text more readable and it makes things a lot easier if we change the URLs for our API website, for example. (Which we did last year.)
https://github.com/dart-lang/site-shared/wiki/Writing-for-Dart-and-Flutter-websites

File: src/docs/cookbook/animation/page-route-animation.md [hunk 2]
```diff
@@ -0,0 +1,277 @@
...
+
+To start, use a
+[PageRouteBuilder](https://api.flutter.dev/flutter/widgets/PageRouteBuilder-class.html)
+to create a [Route](https://api.flutter.dev/flutter/widgets/Route-class.html).
+`PageRouteBuilder` has two callbacks, one to build the content of the route
+(`pageBuilder`), and one to build the route's transition (`transitionsBuilder`).
+
+{{site.alert.note}}
+
+Note: The `child` parameter in transitionsBuilder is the widget returned from
```
Comment (sfshaza2) [inline src/docs/cookbook/animation/page-route-animation.md]: Delete the empty line above and below the note text. We indent notes by two chars, so:

{{site.alert.note}}
  The `child` parameter …
{{site.alert.end}}

Also, the site.alert macro inserts the “Note:” text, so you can delete that.

File: src/docs/cookbook/animation/page-route-animation.md [hunk 3]
```diff
@@ -0,0 +1,277 @@
...
+```
+
+# 2. Create a Tween
+
+To make the new page animate in from the bottom, it should animate from
+`Offset(0,1)` to `Offset(0, 0)` (usually defined using the `Offset.zero`
+constructor). In this case, the Offset is a 2D vector for the
+[FractionalTranslation](https://api.flutter.dev/flutter/widgets/FractionalTranslation-class.html)
+widget. Setting the `dy` argument to 1 represents a vertical translation one
+full height of the page.
```
Comment (sfshaza2) [inline src/docs/cookbook/animation/page-route-animation.md]: Nice addition!

Comment (johnpryan) [reply src/docs/cookbook/animation/page-route-animation.md]: thanks!

**Comment by johnpryan** [comment]
@sfshaza2 thanks for the review - this is ready for another look.

File: src/docs/cookbook/animation/page-route-animation.md (hunk 1 above)
Comment (johnpryan) [reply src/docs/cookbook/animation/page-route-animation.md]: 👍 fixed

File: src/docs/cookbook/animation/page-route-animation.md (hunk 2 above)
Comment (johnpryan) [reply src/docs/cookbook/animation/page-route-animation.md]: 👍 fixed

**Pull request merged**